### PR TITLE
oio: create a shared RedisConn object

### PR DESCRIPTION
**Pull request by murlock**
It will help to manage StrictRedis or Sentinel on various
daemons (account/backend.py and container_streaming at this time)

### CODE REVIEW

File: oio/container_streaming/app.py [hunk 1]
```diff
@@ -320,7 +321,7 @@ def close(self):
             self.logger.info("data not all consumed")
 
 
-class ContainerStreaming(object):
+class ContainerStreaming(RedisConn):
```
Comment (fvennetier) [inline oio/container_streaming/app.py]: Could you rename that? This looks more like a container backup service than like a container "streaming". And I think this does not deserve a package of its own, you should rename the file and move it into the `oio.container` package.

File: oio/container_streaming/app.py [hunk 2]
```diff
@@ -472,7 +471,7 @@ def _do_get(self, req, account, container):
         if 'Range' not in req.headers:
             # TODO: instead expanding blocks to a full list,
             # use [first_block - last_block]
-            tar = TarStreaming(self.conn, account, container, range(blocks),
+            tar = TarStreaming(self.proxy, account, container, range(blocks),
```
Comment (fvennetier) [inline oio/container_streaming/app.py]: `TarStreaming` is also a poor name.

File: oio/container_streaming/app.py [hunk 3]
```diff
@@ -592,7 +591,7 @@ def on_restore(self, req):
             return Response("Not supported", 405)
 
         try:
-            self.conn.container_show(account, container)
+            self.proxy.container_show(account, container)
```
Comment (murlock) [inline oio/container_streaming/app.py]: Done

File: oio/container_streaming/app.py [hunk 4]
```diff
@@ -563,17 +562,17 @@ def read(size):
...
                 else:
-                    self.conn.object_create(account, container,
+                    self.proxy.object_create(account, container,
                                             obj_name=inf.name,
                                             data=read(inf.size))
                     if inf.size % BLOCKSIZE:
                         read(BLOCKSIZE - inf.size % BLOCKSIZE)
                     if hdrs:
-                        self.conn.object_update(account, container, inf.name,
+                        self.proxy.object_update(account, container, inf.name,
```
Comment (murlock) [inline oio/container_streaming/app.py]: Done

File: oio/container_streaming/app.py [hunk 5]
```diff
@@ -563,17 +562,17 @@ def read(size):
                 if inf.name == CONTAINER_PROPERTIES:
                     assert not hdrs, "invalid sequence in TAR"
                     hdrs = json.loads(read(inf.size))
-                    self.conn.container_update(account, container, hdrs)
+                    self.proxy.container_update(account, container, hdrs)
```
Comment (murlock) [inline oio/container_streaming/app.py]: Done
We could emit a info log when using deprecated stuff (for developers only ?)

File: oio/container_streaming/app.py (hunk 2 above)
Comment (murlock) [inline oio/container_streaming/app.py]: is OioTarStreaming ok ?

File: oio/container_streaming/app.py [hunk 6]
```diff
@@ -353,18 +352,18 @@ def generate_oio_map(self, account, container):
...
 
         map_objs = []
         start_block = 0
 
-        meta = self.conn.container_show(account, container)
+        meta = self.proxy.container_show(account, container)
         if meta['properties']:
             # create special file to save properties of container
-            tar = OIOTarEntry(self.conn, account, container,
+            tar = OIOTarEntry(self.proxy, account, container,
```
Comment (murlock) [inline oio/container_streaming/app.py]: Done

File: oio/container_streaming/app.py (hunk 1 above)
Comment (murlock) [inline oio/container_streaming/app.py]: Will be done in a separate commit

File: oio/container_streaming/app.py (hunk 2 above)
Comment (fvennetier) [inline oio/container_streaming/app.py]: The word "streaming" is bothering me. This class is not a "streaming". It may be a "stream". In the docstring you say this class acts as a "File Object", and each instance is dedicated to a single container, why not `ContainerTarFile` or `ContainerTarStream`?

File: oio/container_streaming/app.py (hunk 5 above)
Comment (fvennetier) [inline oio/container_streaming/app.py]: I added some `DeprecationWarning`s in #1053.
